### PR TITLE
FEATURE: revamped wizard

### PR DESCRIPTION
**Pull request by techAPJ**
https://user-images.githubusercontent.com/5732281/178760504-91e2eb42-c617-46a4-b81b-8a21dbb5f370.mov

### CODE REVIEW

File: config/locales/server.en.yml [hunk 1]
```diff
@@ -5019,25 +4957,45 @@ en:
...
-            description: "A square version of your logo. Shown at the top left of your site when scrolling down, in the browser, and when sharing on social platforms. Ideally larger than 512x512."
-
-      icons:
-        title: "Icons"
-        fields:
+            description: "A square version of your logo. Shown at the top left when scrolling down, and when sharing on social platforms. Ideally at least 512x512."
           favicon:
             label: "Browser Icon"
-            description: "Icon image used to represent your site in web browsers that looks good at small sizes. Recommended image extensions are PNG or JPG. We'll use the square logo by default."
+            description: "Icon used to represent your site in web browsers that looks good at small sizes. PNG or JPG is reccomended. Square logo is used by default."
```
Comment (CvX) [inline config/locales/server.en.yml:4971]:
```suggestion
            description: "Icon used to represent your site in web browsers that looks good at small sizes. PNG or JPG is recomended. Square logo is used by default."
```

File: app/assets/javascripts/wizard/addon/components/homepage-preview.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import {
   darkLightDiff,
 } from "wizard/lib/preview";
 
-export default createPreviewComponent(659, 320, {
+export default createPreviewComponent(342, 322, {
```
Comment (CvX) [inline app/assets/javascripts/wizard/addon/components/homepage-preview.js:7]: Haven't tried it locally - why such big dimensions change? 😃

Comment (techAPJ) [reply app/assets/javascripts/wizard/addon/components/homepage-preview.js:7]: The new wizard has reduced the height of the content box which resulted in small preview window.

Old:

Screen Shot 2022-07-25 at 12 30 28

New:

Screen Shot 2022-07-25 at 12 30 58

File: config/locales/server.en.yml (hunk 1 above)
Comment (techAPJ) [reply config/locales/server.en.yml]: Pushed the fix, thanks!

File: app/assets/javascripts/wizard/addon/templates/wizard.hbs [hunk 3]
```diff
@@ -3,44 +3,32 @@
...
-      {{!-- Load all font styles --}}
-      <div class="preloaded-font-styles">
-        {{#each this.fontClasses as |fontClass|}}
-          <span class={{fontClass}}>&nbsp;</span>
-          <strong class={{fontClass}}>&nbsp;</strong>
-        {{/each}}
-      </div>
-    </div>
+  <div class="discourse-logo">
+    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 385 104">
```
[the remark this answers is not in the document]
Comment (techAPJ) [reply app/assets/javascripts/wizard/addon/templates/wizard.hbs:7]: I tried moving the svg block into image but we are using id/class in the svg to style it via CSS, and we're also filling in the color via CSS variable.

[the remark this answers is not in the document]
Comment (CvX) [reply app/assets/javascripts/wizard/addon/templates/wizard.hbs:7]: K, so maybe extract it into a template-only component? 😃 Anything to keep that large blob of svg out of  here 😛

File: lib/wizard/builder.rb [hunk 4]
```diff
@@ -37,108 +47,33 @@ def build
...
-          updater.ensure_changed(:title)
-
-          if updater.errors.blank?
-            updater.apply_settings(:title, :site_description, :short_site_description)
-          end
-        end
-      end
-
-      @wizard.append_step('introduction') do |step|
-        introduction = IntroductionUpdater.new(@wizard.user)
```
Comment (OsamaSayegh) [inline lib/wizard/builder.rb:55]: Do we need to delete `IntroductionUpdater` and its spec file? As far as I can see this is the only spot where it's used.

File: spec/lib/wizard/wizard_builder_spec.rb [hunk 5]
```diff
@@ -122,88 +114,4 @@ class GlobalPathInstance
...
-
-        expect(introduction_step.disabled).to be_nil
-      end
-    end
-  end
-
-  context 'privacy step' do
-    let(:privacy_step) { wizard.steps.find { |s| s.id == 'privacy' } }
-
-    it 'should set the right default value for the fields' do
```
Comment (OsamaSayegh) [inline spec/lib/wizard/wizard_builder_spec.rb:182]: I think it'd be useful to keep this test and update it so it asserts that the new fields (e.g. `invite_only`, `must_approve_users` etc.) have the right default values (unless this is already tested elsewhere).

File: app/assets/javascripts/wizard/addon/components/homepage-preview.js [hunk 6]
```diff
@@ -361,7 +359,9 @@ export default createPreviewComponent(659, 320, {
 
   renderLatest(ctx, colors, font, width, height) {
     const rowHeight = height / 6.6;
-    const textColor = darkLightDiff(colors.primary, colors.secondary, 50, 50);
+    const textColor = colors.primary_medium // accounts for hard-set color variables in solarized themes
+      ? colors.primary_medium
+      : darkLightDiff(colors.primary, colors.secondary, 50, 50);
```
Comment (OsamaSayegh) [inline app/assets/javascripts/wizard/addon/components/homepage-preview.js:364]:
```suggestion
    // accounts for hard-set color variables in solarized themes
    const textColor = colors.primary_medium || darkLightDiff(colors.primary, colors.secondary, 50, 50);
```

(you may need to run prettier)

File: app/assets/javascripts/wizard/addon/components/homepage-preview.js [hunk 7]
```diff
@@ -370,12 +370,9 @@ export default createPreviewComponent(659, 320, {
...
       ctx.beginPath();
-      ctx.strokeStyle = darkLightDiff(
-        colors.primary,
-        colors.secondary,
-        90,
-        -75
-      );
+      ctx.strokeStyle = colors.primary_low // accounts for hard-set color variables in solarized themes
+        ? colors.primary_low
+        : darkLightDiff(colors.primary, colors.secondary, 90, -75);
```
Comment (OsamaSayegh) [inline app/assets/javascripts/wizard/addon/components/homepage-preview.js:375]:
```suggestion
      // accounts for hard-set color variables in solarized themes
      ctx.strokeStyle = colors.primary_low || darkLightDiff(colors.primary, colors.secondary, 90, -75);
```

(you may need to run prettier)

File: app/assets/javascripts/wizard/addon/templates/wizard.hbs (hunk 3 above)
[the remark this answers is not in the document]
Comment (techAPJ) [reply app/assets/javascripts/wizard/addon/templates/wizard.hbs]: Sure, done!

**Comment by nattsw** [comment]
Just tested it out again:

The logo step is good now.
Screenshot 2022-07-25 at 6 12 39 PM

But a small one here occurred.

Screenshot 2022-07-25 at 6 13 54 PM

Could you modify the CSS in this PR, @techAPJ ? And we should be good

File: app/assets/javascripts/wizard/addon/components/wizard-step.js [hunk 8]
```diff
@@ -9,36 +9,45 @@ import { action } from "@ember/object";
...
-  @discourseComputed("step.index")
-  showQuitButton: (index) => index === 0,
-
   @discourseComputed("step.displayIndex", "wizard.totalSteps")
   showNextButton: (current, total) => current < total,
 
-  @discourseComputed("step.displayIndex", "wizard.totalSteps")
-  showDoneButton: (current, total) => current === total,
+  @discourseComputed("step.id", "step.displayIndex", "wizard.totalSteps")
+  showDoneButton: (step, current, total) => {
```
[the remark this answers is not in the document]
Comment (techAPJ) [reply app/assets/javascripts/wizard/addon/components/wizard-step.js]: No specific reason. I copied the existing code which resulted in using old syntax. I just updated all the properties to use new syntax.

File: lib/wizard/builder.rb (hunk 4 above)
Comment (techAPJ) [reply lib/wizard/builder.rb:55]: I was planning to delete those files in follow-up PR, but I deleted them now here.

File: spec/serializers/wizard_serializer_spec.rb [hunk 9]
```diff
@@ -43,14 +43,11 @@
...
-      expect(steps.first['id']).to eq('locale')
-      expect(steps.last['id']).to eq('finished')
+      expect(steps.first['id']).to eq('introduction')
+      expect(steps.last['id']).to eq('corporate')
 
       privacy_step = steps.find { |s| s['id'] == 'privacy' }
       expect(privacy_step).to_not be_nil
-
-      privacy_field = privacy_step['fields'].find { |f| f['id'] == 'privacy' }
-      expect(privacy_field['choices'].find { |c| c['id'] == 'open' }).to_not be_nil
```
[the remark this answers is not in the document]
Comment (techAPJ) [reply spec/serializers/wizard_serializer_spec.rb]: Sure, added tests for these.

File: spec/lib/wizard/wizard_builder_spec.rb (hunk 5 above)
Comment (techAPJ) [reply spec/lib/wizard/wizard_builder_spec.rb:182]: Added back and updated the test.

**Comment by techAPJ** [comment]
Thanks for the review @CvX, @OsamaSayegh, @nattsw! I've addressed all the feedback here. Let me know if the PR can be further improved.

**Review by OsamaSayegh — APPROVED** [review]
LGTM!

**Review by nattsw — APPROVED** [review]

**Pull request merged**